BrewMesModulesParent/api/src/main/java/com/brewmes/api/MachineController.java

```diff
@@ -0,0 +1,105 @@
+package com.brewmes.api;
+
+import com.brewmes.common.entities.Connection;
+import com.brewmes.common.services.IMachineService;
+import com.brewmes.common.util.Command;
+import com.brewmes.common.util.Products;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Optional;
+
+@RestController
+@RequestMapping(value = "/api/machine")
+public class MachineController {
+
+    @Autowired(required = false)
+    private IMachineService machineService;
+
+
+    @GetMapping
+    public ResponseEntity<Object> getConnections() {
+        return new ResponseEntity<>(machineService.getConnections(), HttpStatus.OK);
+    }
+
+    @GetMapping(value = "/{id}")
+    public ResponseEntity<Object> getConnection(@PathVariable("id") String id) {
+        Optional<Connection> connection = Optional.ofNullable(machineService.getConnection(id));
+        if (connection.isPresent()) {
+            return new ResponseEntity<>(connection.get(), HttpStatus.OK);
+        } else {
+            return new ResponseEntity<>("Sorry I don't know that machine :(", HttpStatus.NOT_FOUND);
+        }
+    }
+
+    @DeleteMapping(value = "/{id}")
+    public ResponseEntity<String> removeConnection(@PathVariable("id") String id) {
+        if (machineService.removeConnection(id)) {
+            return new ResponseEntity<>("Machine is removed", HttpStatus.OK);
+        } else {
+            return new ResponseEntity<>("Could not remove machine", HttpStatus.NOT_ACCEPTABLE);
+        }
+    }
+
+    @PostMapping(value = "/{id}")
+    public ResponseEntity<String> addConnection(@PathVariable("id") String id, @RequestBody String name) {
+        if (machineService.addConnection(id, name)) {
+            return new ResponseEntity<>("Machine is added", HttpStatus.OK);
+        } else {
+            return new ResponseEntity<>("Could not add machine", HttpStatus.NOT_ACCEPTABLE);
+        }
+    }
+
+    @PutMapping(value = "/{id}/control")
+    public ResponseEntity<String> controlMachine(@PathVariable("id") String id, @RequestBody String command) {
+        if (machineService.controlMachine(Command.valueOf(command.toUpperCase()), id)) {
+            return new ResponseEntity<>("Command successful", HttpStatus.OK);
+        } else {
+            return new ResponseEntity<>("Command unsuccessful", HttpStatus.NOT_ACCEPTABLE);
```

Comment:
Evt. 400 Bad Request. Hvis man læser om 406 Not Acceptable lyder den rimelig omfattende